eventlet/support/afinet.py

```diff
@@ -0,0 +1,46 @@
+"""
+Test availabilty of AF_INET6 on a machine
+
+Uses socket API to determine configuration
+"""
+
+from socket import *
+
+def check_ipv6_lo_addr():
+    try:
+        sock = socket(AF_INET6, SOCK_STREAM)
+        sock.bind(('::1', 0))
+        return True
+    except:
+        return False
+
+def check_ipv6_hosts_file():
+    if not check_ipv6_lo_addr():
+        return False
+    addrinfo = getaddrinfo(gethostname(), 0)
+    if (addrinfo[0][0] == AF_INET6):
```

Comment:
- `if (a) -> if a`, Python doesn't need parens.
- `if a: return True else: return False -> return bool(a)`. In this case it's even simpler `return ...==AFINET6`.
- It needs bounds check. There may be no elements in list.
- This check only considers first address. `not all(...)` construct is consistent with promise of "for machines with no IPv4".